Straight_skeleton_2/test/Straight_skeleton_2/CMakeLists.txt

```diff
@@ -5,23 +5,34 @@ cmake_minimum_required(VERSION 3.1...3.23)
 project(Straight_skeleton_2_Tests)
 
 find_package(CGAL REQUIRED COMPONENTS Qt6 Core)
+find_package(LEDA QUIET)
 
 include_directories(BEFORE "include")
 
-# create a target per cppfile
-file(
-  GLOB cppfiles
-  RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
-  ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
-foreach(cppfile ${cppfiles})
-  create_single_source_cgal_program("${cppfile}")
-endforeach()
+create_single_source_cgal_program("issue4533.cpp")
+create_single_source_cgal_program("issue4684.cpp")
+create_single_source_cgal_program("test_sls.cpp")
+create_single_source_cgal_program("test_sls_previous_issues.cpp")
+create_single_source_cgal_program("test_sls_traits.cpp")
+create_single_source_cgal_program("test_straight_skeleton_copy.cpp")
 
 if(CGAL_Qt6_FOUND)
   target_link_libraries(issue4684 PUBLIC CGAL::CGAL_Basic_viewer)
-  target_link_libraries(issue7149 PUBLIC CGAL::CGAL_Basic_viewer)
-  target_link_libraries(issue7284 PUBLIC CGAL::CGAL_Basic_viewer)
   target_link_libraries(test_sls_previous_issues PUBLIC CGAL::CGAL_Basic_viewer)
-  target_link_libraries(test_sls_offset PUBLIC CGAL::CGAL_Basic_viewer)
-  target_link_libraries(test_sls_weighted_polygons_with_holes PUBLIC CGAL::CGAL_Basic_viewer)
+endif()
+
+if (CGAL_Core_FOUND OR LEDA_FOUND)
+  create_single_source_cgal_program("test_sls_offset.cpp")
+  create_single_source_cgal_program("issue7284.cpp")
+  create_single_source_cgal_program("test_sls_simple.cpp")
+  create_single_source_cgal_program("test_sls_weighted_polygons_with_holes.cpp")
+  create_single_source_cgal_program("issue7149.cpp")
+  if(CGAL_Qt6_FOUND)
+    target_link_libraries(issue7149 PUBLIC CGAL::CGAL_Basic_viewer)
+    target_link_libraries(issue7284 PUBLIC CGAL::CGAL_Basic_viewer)
+    target_link_libraries(test_sls_offset PUBLIC CGAL::CGAL_Basic_viewer)
+    target_link_libraries(test_sls_weighted_polygons_with_holes PUBLIC CGAL::CGAL_Basic_viewer)
+  endif()
+else()
+  message("NOTICE: Some test require CGAL_Core (or LEDA), and will not be compiled.")
```

Comment:
Isn't there a more flexible way possible?
(see e.g. https://github.com/doxygen/doxygen/blob/master/examples/CMakeLists.txt and look for BASIC_EXAMPLES. still a list but a bit more flexible, maybe something is possible here as well...)